src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -245,9 +252,44 @@ public void setWebhookRegistered(boolean webhookRegistered) {
     protected void retrieve(@CheckForNull SCMSourceCriteria criteria, SCMHeadObserver observer,
                             @CheckForNull SCMHeadEvent<?> event,
                             TaskListener listener) throws IOException, InterruptedException {
+        if (event == null) {
+            //on manual scans & creation of jenkins jobs, we call out to bbs and fetch a list of open prs and
+            // sync up our local pullRequestStore
+            DescriptorImpl descriptor = (DescriptorImpl) getDescriptor();
+            List<BitbucketPullRequest> bbsPullRequests = fetchOpenPullRequestsFromBbsInstance(descriptor);
+            String serverId = getServerId();
+            if (serverId != null) {
+                descriptor.getPullRequestStore().refreshStore(getProjectKey(), getRepositorySlug(), serverId, bbsPullRequests);
+            }
+        }
         gitSCMSource.accessibleRetrieve(criteria, observer, event, listener);
     }
 
+    private List<BitbucketPullRequest> fetchOpenPullRequestsFromBbsInstance(DescriptorImpl descriptor) {
```

Comment:
I think you should change the API here, return a stream instead. That way you can keep efficiency high. 
Rather than `stream->list->stream->list` which is the current model, we would get `stream->list` which is better